### PR TITLE
Update benchmark scripts, use uv

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Use uv instead of poetry, use `ruff check` vs `ruff`

Related https://github.com/astral-sh/ruff/issues/15116

## Test Plan

I ran all the benchmark scripts locally. I couldn't get pylint to work

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -467,7 +467,7 @@ cargo build --release && hyperfine --warmup 10 \
   "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache -e --select W505,E501"
 ```
 
-You can run `poetry install` from `./scripts/benchmarks` to create a working environment for the
+You can run `uv venv --project ./scripts/benchmarks`, activate the venv and then run `uv venv --project ./scripts/benchmarks` to create a working environment for the
```
Comment (dhruvmanila) [inline CONTRIBUTING.md:470]: These two commands are the same but the text says to run the first and then the second (same) command.

File: scripts/.python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+<3.12
```
Comment (dhruvmanila) [inline scripts/.python-version:1]: What's the reason for this?

File: scripts/benchmarks/README.md [hunk 3]
```diff
@@ -4,10 +4,10 @@ Utilities for benchmarking Ruff.
 
 ## Getting Started
 
-Run `./scripts/benchmarks/run.sh` to clone the benchmarking target (CPython).
+Run `./scripts/benchmarks/setup.sh` to clone the benchmarking target (CPython).
 
-If you're looking to benchmark Ruff against other tools, you'll also need to run `poetry install`
-to create a virtual environment with the required dependencies.
+If you're looking to benchmark Ruff against other tools, you'll also need to run `uv venv --project ./scripts/benchmarks`, 
+activate the venv and finally `uv sync ./scripts/benchmarks` to create a virtual environment with the required dependencies.
```
Comment (dhruvmanila) [inline scripts/benchmarks/README.md:10]: The `uv sync ./scripts/benchmarks` doesn't work. Did you mean to use `--project` flag? Although, even after that it throws an error:
```console
$ uv sync --project ./scripts/benchmark
Using CPython 3.8.18 interpreter at: /Users/dhruv/.local/bin/python3.8
error: The Python request from `scripts/.python-version` resolved to Python 3.8.18, which is incompatible with the project's Python requirement: `>=3.11`. Use `uv python pin` to update the `.python-version` file to a compatible version.
```

Removing the `./scripts/.python-version` does make the command run but that creates the virtual environment in `./scripts` directory while the `uv venv --project ./scripts/benchmarks` command creates the environment in `./scripts/benchmarks` directory.

File: scripts/benchmarks/run_plugins.sh [hunk 4]
```diff
@@ -4,40 +4,40 @@
...
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PD --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PGH --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PLC --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PLE --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PLR --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PLW --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select PIE --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select COM --exit-zero" \
+  "./target/release/ruff check ./crates/ruff_linter/resources/test/cpython/ --no-cache --silent --extend-select R --exit-zero"
```
Comment (dhruvmanila) [inline scripts/benchmarks/run_plugins.sh:43]: I think this should be `--extend-select RUF`

File: scripts/.python-version (hunk 2 above)
Comment (MichaReiser) [reply scripts/.python-version:1]: It's to match what we had in the `pyproject.toml` but I'm not sure if it even works. I'll removee

**Review by dhruvmanila — APPROVED** [review]
Thank you

**Comment by MichaReiser** [comment]
Thank you for your careful review and sorry for all my silly mistakes.

**Pull request merged**